bgpd/bgp_vty.c

```diff
@@ -10943,6 +10943,58 @@ DEFUN (no_bgp_segment_routing_srv6,
 	return CMD_SUCCESS;
 }
 
+DEFPY (bgp_srv6_encap_behavior,
+       bgp_srv6_encap_behavior_cmd,
+       "[no$no] encap-behavior <H_Encaps|H_Encaps_Red>$encap_behavior",
+       NO_STR
+       "Configure SRv6 encap mode\n"
+       "H.Encaps\n"
+       "H.Encaps.Red\n")
+{
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+	enum srv6_headend_behavior srv6_encap_behavior;
+	struct bgp *bgp_inst;
+	struct listnode *node;
+
+	bgp = bgp_get_default();
+	if (!bgp)
+		return CMD_SUCCESS;
+
+	if (strmatch(encap_behavior, "H_Encaps_Red"))
+		srv6_encap_behavior = SRV6_HEADEND_BEHAVIOR_H_ENCAPS_RED;
+	else
+		srv6_encap_behavior = SRV6_HEADEND_BEHAVIOR_H_ENCAPS;
+
+	if (no && ((srv6_encap_behavior != bgp->srv6_encap_behavior) ||
+		   (srv6_encap_behavior == SRV6_HEADEND_BEHAVIOR_H_ENCAPS)))
+		return CMD_SUCCESS;
+	else if (no == NULL && srv6_encap_behavior == bgp->srv6_encap_behavior)
+		return CMD_SUCCESS;
+
+	if (no)
+		bgp->srv6_encap_behavior = SRV6_HEADEND_BEHAVIOR_H_ENCAPS;
+	else
+		bgp->srv6_encap_behavior = srv6_encap_behavior;
+
+	for (ALL_LIST_ELEMENTS_RO(bm->bgp, node, bgp_inst)) {
+		/*
+		 * Stop a crash, more work is needed
```

Comment:
What kind of crash?

Reply:
The same that is here : https://github.com/FRRouting/frr/blob/master/bgpd/bgpd.c#L273 .
I never saw such a crash with srv6, because I guess the setup I use do not have such many srv6 routes.
Even with many routes, I never fell on this.